src/main/webapp/app/overview/course-card.component.ts

```diff
@@ -60,7 +60,11 @@ export class CourseCardComponent implements OnChanges {
     ngOnChanges() {
         if (this.course.exercises && this.course.exercises.length > 0) {
             this.exerciseCount = this.course.exercises.length;
-            const nextExercises = this.exerciseService.getNextExercisesForDays(this.course.exercises);
+            // sets the exercise with the earliest due date which is not yet successful (no 100%)
+            const nextExercises = this.exerciseService.getNextExercisesForDays(
+                this.course.exercises!.filter((exercise: Exercise) => !exercise.studentParticipations?.[0]?.submissions?.[0]?.results?.[0]?.successful),
```

Comment:
can you extract a variable here? otherweise the code gets too complex in my opinion.

Reply:
Yes, I hope this is ok. 
![image](https://user-images.githubusercontent.com/65814168/117274671-93375280-ae5d-11eb-9281-33be2a1bda54.png)